packages/localize-monetary-amount/src/localize-monetary-amount.ts

```diff
@@ -0,0 +1,980 @@
+ *     https://www.loc.gov/standards/iso639-2/php/English_list.php
+ *
+ * @param {LocaleCode<Raw>} rawLocaleCode
+ *   ISO 631-1 language code with an optional ISO 3166-1 alpha-2 region code,
+ *   separated by a hyphen. Examples: 'en', 'en-gb', 'fr-be'.
+ * @param {CurrencyCode} currencyCode
+ *   ISO 4217 currency code. Examples: 'USD', 'JPY', 'BRL'.
+ * @param {number} amount
+ *   Integer amount in minor currency units. Examples: $1.00 USD and ¥100 JPY
+ *   are both passed as 100.
+ * @returns {string}
+ *   Localized monetary amount
+ */
+export function localizeMonetaryAmount(
+	rawLocaleCode: LocaleCode< Raw >,
+	currencyCode: CurrencyCode,
+	amount: number
+): LocalizedMonetaryAmount {
+	// Validate arguments
+	if ( ! ( typeof amount === 'number' && Number.isInteger( amount ) ) ) {
+		throw 'amount parameter must be an integer.';
+	}
+
+	const localeCode = normalizeLocaleCode( rawLocaleCode );
+
+	return localizeCurrencyWithSchema(
+		Math.sign( amount ),
+		Math.abs( amount ),
+		localeCode,
+		currencyCode,
+		currencyFormattingSchema( localeCode, currencyCode )
+	);
+}
+
+export type LocalizedMonetaryAmount = string;
+
+/**
+ * Representation of the possible currency schemas.
+ */
+enum CurrencyFormat {
+	LocalSymbol_Amount,
+	LocalSymbol_Amount_Code,
+}
+
+/**
+ * Selects a reified formatting style for the given locale and currency.
+ * This style is represented by a slug called a /schema/, and consumed by
+ * localizeCurrencyWithSchema() to construct the formatted string.
+ *
+ * @param {LocaleCode<Normalized>} localeCode
+ *   ISO 631-1 language code with an optional ISO 3166-1 alpha-2 region code,
+ *   separated by a hyphen. Examples: 'en', 'en-gb', 'fr-be'.
+ * @param {CurrencyCode} currencyCode
+ *   ISO 4217 currency code. Examples: 'USD', 'JPY', 'BRL'.
+ * @returns {CurrencyFormat}
+ *   Currency formatting schema slug
+ */
+function currencyFormattingSchema(
+	localeCode: LocaleCode< Normalized >,
+	currencyCode: CurrencyCode
+): CurrencyFormat {
+	switch ( localeCode ) {
+		case 'en-us':
+			// Most common case first
+			if ( currencyCode === 'USD' ) {
+				return CurrencyFormat.LocalSymbol_Amount;
+			}
+
+			if ( hasUniqueLocalSymbol( currencyCode ) ) {
+				return CurrencyFormat.LocalSymbol_Amount;
+			}
+
+			switch ( currencyCode ) {
+				case 'GBP':
+					return CurrencyFormat.LocalSymbol_Amount;
+
+				default:
+					return CurrencyFormat.LocalSymbol_Amount_Code;
+			}
+
+		default:
+			if ( hasUniqueLocalSymbol( currencyCode ) ) {
+				return CurrencyFormat.LocalSymbol_Amount;
+			}
+
+			switch ( currencyCode ) {
+				case 'GBP':
+					return CurrencyFormat.LocalSymbol_Amount;
+
+				// Default to generic verbose format
+				default:
+					return CurrencyFormat.LocalSymbol_Amount_Code;
+			}
+	}
+}
+
+/**
+ * Localize a monetary amount using the given schema slug.
+ *
+ * @param {number} amountSign
+ *   1, 0, or -1, to distinguish positive and negative amounts
```

Comment:
In this library we really do only want the "sign" and not to worry about negative zero since this package only deals with integers. I ended up defining a `const enum` type to represent the sign, so we can define away float weirdness. :)

Reply:
Nice solution 👌